geth/transactions/txqueue_manager.go

```diff
@@ -207,14 +208,21 @@ func (m *Manager) completeTransaction(config *params.NodeConfig, selectedAccount
 	}
 
 	chainID := big.NewInt(int64(config.NetworkID))
-	data := []byte(args.Data)
+	input := args.Input
+	if input == nil {
+		input = args.Data
+	}
+	if input == nil {
+		return hash, errors.New("Missing Input/Data byte array")
```

Comment:
That should be totally fine. Do you get an error if `data` passed to `tx := types.NewTransaction(nonce, toAddr, value, gas, gasPrice, data)` is `nil`?